From 79f96565ad9bd32139c4cf1b58e2adad1b85182b Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Fri, 19 Oct 2018 12:10:17 +0200
Subject: [PATCH] Merge android-9.0.0_r12

commit 1603d1eed5d893f95790eec250d847e592cc1eea
Merge: a2ea9593 01756782
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jul 31 01:09:08 2018 +0000

    Merge cherrypicks of [4658485, 4658486, 4658487, 4657188, 4657275, 4659124, 4659125, 4659141, 4659142, 4659143, 4657276, 4657814, 4658660, 4657277, 4657815, 4657278, 4659144, 4658126, 4658127, 4658128, 4658129, 4658130, 4658131, 4658132, 4658133, 4658134, 4658135, 4658136, 4658137, 4658138, 4659161, 4659162, 4659163, 4659164, 4659145, 4659146, 4659147, 4659148, 4657655, 4657656, 4658139, 4658140, 4659181, 4659182, 4659183, 4657130, 4657131, 4657132, 4657133, 4659165, 4659166, 4659167, 4657657, 4657279, 4659149] into pi-dr1-release

    Change-Id: Ic384d52560848a311a7decfcc171c6a16e72c9cd

commit 01756782a54ae4092a3501c43a5860cc42b7eb47
Author: Yoshiaki Naka <yoshiaki.naka@sony.com>
Date:   Fri Jun 8 10:05:34 2018 +0900

    SecureElementService may be unavailable when NfcService is started

    Currently NfcService attempts to get an interface to
    SecureElementService only in its constructor, so there is no chance to
    do it later again if it fails. Introducing a lazy initialization
    mechanism is required.

    Bug: 111775064
    Bug: 110121800
    Test: Confirmed that NfcService gets SEService even if it failed once.

    Change-Id: I2b3c873d7661cde63af8c9f8c05945c3568f012f
    (cherry picked from commit a9ef080f9964f46d4983bf7637a2c52a6e0eadd2)

Change-Id: I40448737e3a12a7c4c628662654c0821825cf7b4
---
 src/com/android/nfc/NfcService.java | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)
 mode change 100755 => 100644 src/com/android/nfc/NfcService.java

diff --git a/src/com/android/nfc/NfcService.java b/src/com/android/nfc/NfcService.java
old mode 100755
new mode 100644
index d5835ca7..0c37cd80
--- a/src/com/android/nfc/NfcService.java
+++ b/src/com/android/nfc/NfcService.java
@@ -517,6 +517,14 @@ public class NfcService implements DeviceHostListener {
                 Context.SECURE_ELEMENT_SERVICE));
     }
 
+    private boolean isSEServiceAvailable() {
+        if (mSEService == null) {
+            mSEService = ISecureElementService.Stub.asInterface(ServiceManager.getService(
+                    Context.SECURE_ELEMENT_SERVICE));
+        }
+        return (mSEService != null);
+    }
+
     void initSoundPool() {
         synchronized (this) {
             if (mSoundPool == null) {
@@ -2251,7 +2259,7 @@ public class NfcService implements DeviceHostListener {
         }
 
         private void sendOffHostTransactionEvent(byte[] aid, byte[] data, byte[] readerByteArray) {
-            if (mSEService == null || mNfcEventInstalledPackages.isEmpty()) {
+            if (!isSEServiceAvailable() || mNfcEventInstalledPackages.isEmpty()) {
                 return;
             }
 
@@ -2291,7 +2299,7 @@ public class NfcService implements DeviceHostListener {
 
         /* Returns the list of packages that have access to NFC Events on any SE */
         private ArrayList<String> getSEAccessAllowedPackages() {
-            if (mSEService == null || mNfcEventInstalledPackages.isEmpty()) {
+            if (!isSEServiceAvailable() || mNfcEventInstalledPackages.isEmpty()) {
                 return null;
             }
             String[] readers = null;
-- 
2.17.1

